### PR TITLE
#8 Fix unpacking of values for WritingMultipleCoils request.

### PR DESCRIPTION
**Pull request by OrangeTux**

### CODE REVIEW

File: umodbus/functions.py [hunk 1]
```diff
@@ -665,6 +664,41 @@ def __init__(self, starting_address, quantity, byte_count, values):
     def create_from_request_pdu(cls, pdu):
         """ Create instance from request PDU.
 
+        This method requires some clarification regarding the unpacking of
+        the statusses that are being passed to the callbacks.
```
Comment (jaapz) [inline umodbus/functions.py]: > statusses

moet zijn

>statuses

File: umodbus/functions.py [hunk 2]
```diff
@@ -665,6 +664,41 @@ def __init__(self, starting_address, quantity, byte_count, values):
...
+
+        A request pdu contains at least 1 byte representing value for
```
Comment (jaapz) [inline umodbus/functions.py]: > value

moet zijn

> the value

File: umodbus/functions.py [hunk 3]
```diff
@@ -665,6 +664,41 @@ def __init__(self, starting_address, quantity, byte_count, values):
...
+        addresses. When quantity field is set to 1, the LSB contains value for
```
Comment (jaapz) [inline umodbus/functions.py]: >the value

File: umodbus/functions.py [hunk 4]
```diff
@@ -665,6 +664,41 @@ def __init__(self, starting_address, quantity, byte_count, values):
...
+        the starting address. When starting address is 100, quantity and value
+        is 0 value for coil 100 is 0 as you can see in the first row of the
+        table below.
```
Comment (jaapz) [inline umodbus/functions.py]: I don't understand what this sentence is trying to say, please rephrase

File: umodbus/functions.py [hunk 5]
```diff
@@ -665,6 +664,41 @@ def __init__(self, starting_address, quantity, byte_count, values):
...
+
+        When quantity is 2 and value is 0 then value for addresses 100 and 101
+        is both 0, see row 3.
```
Comment (jaapz) [inline umodbus/functions.py]: > is

> are

File: umodbus/functions.py [hunk 6]
```diff
@@ -665,6 +664,41 @@ def __init__(self, starting_address, quantity, byte_count, values):
...
+
+        When quantit is 2 and value is 2, the value of address 100 is 0 and
```
Comment (jaapz) [inline umodbus/functions.py]: quantity

> is

should be 

> are

File: umodbus/functions.py [hunk 7]
```diff
@@ -672,12 +706,18 @@ def create_from_request_pdu(cls, pdu):
...
         fmt = '>' + ('B' * byte_count)
         values = struct.unpack(fmt, pdu[6:])
-        values = [integer_to_binary_list(v) for v in values]
 
-        # Flatten list
-        values = [n for value in values for n in value]
+        res = list()
 
-        return cls(starting_address, quantity, byte_count, values)
+        for i, value in enumerate(values):
```
Comment (jaapz) [inline umodbus/functions.py]: The stuff below seems like magic to me. Could you document why this is happening?

File: umodbus/functions.py [hunk 8]
```diff
@@ -672,12 +706,18 @@ def create_from_request_pdu(cls, pdu):
...
+            padding = 8 if (quantity - (8 * i)) // 8 > 0 else quantity % 8
+            fmt = '{{0:0{padding}b}}'.format(padding=padding)
+
+            # Create binary representation of integer, convert it to a list
+            # and reverse the list.
```
Comment (jaapz) [inline umodbus/functions.py]: Why create the binary representatie, why convert it to a list and why reverse it? I can read the code and see it is doing this, but why is this happening?

File: umodbus/functions.py (hunk 7 above)
Comment (OrangeTux) [inline umodbus/functions.py]: I don't understand what is magic about it. It's an iteration of a list of values like: `[6, 144, 255]`.

File: umodbus/functions.py (hunk 8 above)
Comment (OrangeTux) [inline umodbus/functions.py]: I've updated the docstring and hopefully reasoning is clear now.

**Comment by jaapz** [comment]
LGTM :+1:

**Pull request merged**